extensions/opentelemetry/opentelemetry-async-mutiny/deployment/src/test/java/io/quarkus/opentelemetry/async/mutiny/deployment/WithSpanInterceptorTest.java

```diff
@@ -0,0 +1,301 @@
+        public Uni<String> span() {
+            return createDefaultUni();
+        }
+
+        @WithSpan
+        public Uni<String> slow() {
+            return createDefaultUni(10000);
+        }
+
+        @WithSpan("name")
+        public Uni<String> spanName() {
+            return createDefaultUni();
+        }
+
+        @WithSpan(kind = SERVER)
+        public Uni<String> spanKind() {
+            return createDefaultUni();
+        }
+
+        @WithSpan
+        public Uni<String> spanArgs(@SpanAttribute(value = "arg") String arg) {
+            return createDefaultUni();
+        }
+
+        @Inject
+        SpanChildBean spanChildBean;
+
+        @WithSpan
+        public Uni<String> spanChild() {
+            return spanChildBean.spanChild();
+        }
+
+        @WithSpan
+        public Multi<String> multiSpan() {
+            return createDefaultMulti();
+        }
+
+        @WithSpan
+        public Multi<String> multiSpanConcat() {
+            return Multi.createBy().concatenating().streams(spanChildBean.stream1(), spanChildBean.stream2());
+        }
+
+        @WithSpan
+        public Multi<String> multiSpanMerge() {
+            return Multi.createBy().merging().streams(spanChildBean.stream1(), spanChildBean.stream2());
+        }
+    }
+
+    @ApplicationScoped
+    public static class SpanChildBean {
+
+        @WithSpan
+        public Uni<String> spanChild() {
+            return createDefaultUni();
+        }
+
+        @WithSpan
+        public Multi<String> stream1() {
+            return createDefaultMulti(200);
+        }
+
+        @WithSpan
+        public Multi<String> stream2() {
+            return createDefaultMulti(100);
+        }
+    }
+
+    private static Uni<String> createDefaultUni() {
+        return createDefaultUni(100);
+    }
+
+    private static Uni<String> createDefaultUni(final int delayMs) {
+        return createDefaultUni(getCurrentSpanContext(), delayMs);
+    }
+
+    private static Uni<String> createDefaultUni(final SpanContext context) {
+        return createDefaultUni(context, 100);
+    }
+
+    private static Uni<String> createDefaultUni(final SpanContext context, final int delayMs) {
+        return Uni.createFrom().item("QUARKUS").onItem().delayIt().by(Duration.ofMillis(delayMs)).onItem()
+                .invoke(() -> assertThat(Span.current().getSpanContext()).isEqualTo(context));
+    }
+
+    private static Multi<String> createDefaultMulti() {
+        return createDefaultMulti(100);
+    }
+
+    private static Multi<String> createDefaultMulti(final SpanContext context) {
+        return createDefaultMulti(context, 100);
+    }
+
+    private static Multi<String> createDefaultMulti(final int delayMs) {
+        return createDefaultMulti(getCurrentSpanContext(), delayMs);
+    }
+
+    private static Multi<String> createDefaultMulti(final SpanContext context, final int delayMs) {
+        return Multi.createFrom().items("SUPERSONIC", "SUBATOMIC", "JAVA")
```

Comment:
Both Uni and Multi tests use the "delayIt" operators which use the Quarkus scheduler. 
It would be interesting to test:

- truly async using the duplicated context
- unmanaged thread.